janus_core/janus_types.py

```diff
@@ -0,0 +1,69 @@
+"""
+Module containing types used in Janus-Core.
+"""
+
+from collections.abc import Sequence
+from pathlib import Path, PurePath
+from typing import IO, Literal, Optional, TypedDict, TypeVar, Union
+
+from ase import Atoms
+import numpy as np
+from numpy.typing import NDArray
+
+# General
+
+T = TypeVar("T")
+MaybeList = Union[T, list[T]]
+MaybeSequence = Union[T, Sequence[T]]
+PathLike = Union[str, Path]
+
+
+# ASE Arg types
+
+
+class ASEReadArgs(TypedDict, total=False):
+    """Main arguments for ase.io.read."""
+
+    filename: Union[str, PurePath, IO]
+    index: Union[int, slice, str]
+    format: Optional[str]
+    parallel: bool
+    do_not_split_by_at_sign: bool
+
+
+class ASEWriteArgs(TypedDict, total=False):
+    """Main arguments for ase.io.write."""
+
+    filename: Union[str, PurePath, IO]
+    images: MaybeSequence[Atoms]
+    format: Optional[str]
+    parallel: bool
+    append: bool
+
+
+class ASEOptArgs(TypedDict, total=False):
+    """Main arugments for ase optimisers."""
+
+    restart: Optional[bool]
+    logfile: Optional[PathLike]
+    trajectory: PathLike
```

Comment:
```suggestion
    trajectory: Optional[str]
```

I don't think this actually accepts `pathlib.Path` etc., only `str`, unlike ase.io.read/write.